### PR TITLE
JDKResource priorities

### PR DESCRIPTION
**Pull request by alexeybakhtin**
Added priority enumeration for the JDK resources
It will allow better handling the order of checkpoint notifications for dependent resources

---------
### Progress
- [x] Change must not contain extraneous whitespace

### Reviewers
 * [Anton Kozlov](https://openjdk.java.net/census#akozlov) (@AntonKozlov - **Reviewer**) ⚠️ Review applies to 58f23f6739f050f46bc031843355451bf5d624ef
 * [Dan Heidinga](https://openjdk.java.net/census#heidinga) (@DanHeidinga - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/crac pull/8/head:pull/8` \
`$ git checkout pull/8`

Update a local copy of the PR: \
`$ git checkout pull/8` \
`$ git pull https://git.openjdk.java.net/crac pull/8/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8`

View PR using the GUI difftool: \
`$ git pr show -t 8`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/crac/pull/8.diff">https://git.openjdk.java.net/crac/pull/8.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/jdk/internal/crac/JDKResource.java [hunk 1]
```diff
@@ -29,5 +29,23 @@
...
+     * Most resources should use NORMAL.
+     * Other priorities define sequence of
+     * checkpoint notification for dependent resources
+     */
+    enum Priority {
+        /**
+         * Priority of the
+         * sun.nio.ch.EPollSelectorImpl resource
+         */
+        EPOLLSELECTOR,
```
Comment (AntonKozlov) [inline src/java.base/share/classes/jdk/internal/crac/JDKResource.java:43]: This uncovers the future mess we'll have with manually setting the order, similar to the mutex order in the Hotspot. However, I like this central place to control them. This approach is by no means worse than the existing code but highlights a point of growth.

**Review by AntonKozlov — APPROVED** [review]

File: src/java.base/share/classes/jdk/internal/crac/JDKContext.java [hunk 2]
```diff
@@ -35,7 +35,7 @@
     static class ContextComparator implements Comparator<Map.Entry<JDKResource, Void>> {
         @Override
         public int compare(Map.Entry<JDKResource, Void> o1, Map.Entry<JDKResource, Void> o2) {
-            return o2.getKey().getPriority() - o1.getKey().getPriority();
+            return o1.getKey().getPriority().ordinal() - o2.getKey().getPriority().ordinal();
```
Comment (DanHeidinga) [inline src/java.base/share/classes/jdk/internal/crac/JDKContext.java:38]: Enums are already comparable so we can use the built in compareTo method here but this isn't critical and shouldn't block merging

```suggestion
            return o1.getKey().getPriority().compareTo(o2.getKey().getPriority());
```

**Review by DanHeidinga — APPROVED** [review]

File: src/java.base/share/classes/jdk/internal/crac/JDKContext.java (hunk 2 above)
Comment (alexeybakhtin) [reply src/java.base/share/classes/jdk/internal/crac/JDKContext.java]: Hi @DanHeidinga !
Thank you for review. Yes, compareTo looks better. Updated.

**Review by DanHeidinga — APPROVED** [review]
lgtm

**Pull request closed without merging**